lib/rules/no-useless-escape.js

```diff
@@ -95,10 +114,12 @@ module.exports = function(context) {
             return;
         }
 
-        (node.raw.match(/\\[^\d]/g) || []).forEach(validate, {
-            "escapes": nodeEscapes,
-            "node": node
-        });
+        while ((match = pattern.exec(node.raw))) {
```

Comment:
@BYK What context object is created in each iteration? Assuming the pattern has the "g" flag, this is a known pattern for evaluating the code one match at a time, and as far as I'm aware no waste is created.